src/main/java/com/artipie/api/RepositoryRest.java

```diff
@@ -110,6 +116,28 @@ private void getRepo(final RoutingContext context) {
         }
     }
 
+    /**
+     * Checks if repository settings exist.
+     * @param context Routing context
+     */
+    private void existRepo(final RoutingContext context) {
+        final RepositoryName rname = new RepositoryName.FromRequest(context, this.layout);
+        final ReservedNamesVerifier reserved = new ReservedNamesVerifier(rname);
+        final SettingsDuplicatesVerifier duplicates =
+            new SettingsDuplicatesVerifier(rname, this.crs);
+        final ExistenceVerifier existence = new ExistenceVerifier(rname, this.crs);
+        final Validator validator = new Validator.All(
```

Comment:
To avoid code duplication, suggest adding an interface `Verifier` with methods `valid` and `message`.
Use it as a parameter for `Validator.validator`.